src/gazebo_mavlink_interface.cpp

```diff
@@ -769,9 +769,10 @@ void GazeboMavlinkInterface::SendGroundTruth()
   hil_state_quat.true_airspeed = (model_->GetWorldLinearVel() -  wind_vel_).GetLength() * 100;
 #endif
 
-  hil_state_quat.xacc = accel_true_b.X() * 1000;
-  hil_state_quat.yacc = accel_true_b.Y() * 1000;
-  hil_state_quat.zacc = accel_true_b.Z() * 1000;
+  ignition::math::Vector3d accel_true_b_in_ned = q_nb.RotateVector(accel_true_b);
+  hil_state_quat.xacc = accel_true_b_in_ned.X() * 1000;
+  hil_state_quat.yacc = accel_true_b_in_ned.Y() * 1000;
+  hil_state_quat.zacc = accel_true_b_in_ned.Z() * 1000;
```

Comment:
Would it be possible to save one rotation https://github.com/PX4/PX4-SITL_gazebo/pull/827/files#diff-b6e0a79395aaa68135ca1c7ed7c931289bc2f154042330e4f3a7a00ed99ffba2R734 ?

Reply:
Both FLU and FRD are in body frame, so a rotation from FRD to NED is necessary.
https://github.com/PX4/PX4-SITL_gazebo/blob/4c27fc7dd659c262257abeea2308e2f1c54e9029/include/common.h#L213-L237

Reply:
Yes, since we are doing two rotations now, while we can get away with one.

Reply:
> Yes, since we are doing two rotations now, while we can get away with one.

Then I recommend to rename `accel_true_b` (`accel_true_ned` with `ned` for `NED ref frame`?), since there is a `accel_b` in FRD body frame.

Reply:
The `accel_true_ned` will only use one rotation now.
https://github.com/PX4/PX4-SITL_gazebo/pull/827/files#diff-b6e0a79395aaa68135ca1c7ed7c931289bc2f154042330e4f3a7a00ed99ffba2